lib/reply.js

```diff
@@ -107,7 +107,12 @@ Reply.prototype.send = function (payload) {
     if (hasContentType === false || contentType.indexOf('charset') === -1) {
       this[kReplyHeaders]['content-type'] = CONTENT_TYPE.JSON
     }
-    payload = serialize(this.context, payload, this.res.statusCode)
+    try {
+      payload = serialize(this.context, payload, this.res.statusCode)
+    } catch (error) {
+      onErrorHook(this, error, onSendHook)
+      return
+    }
```

Comment:
Do you suggest to reset `sent` in the catch block (instead of call `onErrorHook`) or set `sent = true` just a little before to call `onSendHook`?
The first approach keeps the performance degradation, so the second one appears better to me, but I haven't a deep knowledge of Fastify codebase and I cannot evaluate if this could have some side effects...

Reply:
This change is not really related to performance, but correctness. We should not be invoking here `onErrorHook`, but rather bubble the error to the user. We'll focus on optimizing after that change.

Note that the call to `this[kReplySerializer](payload)` would need to be protected in a similar way.

Reply:
I changed the way `sent` is set, to cover all cases without try-catches. Benchmark updated.